### PR TITLE
Wrap too long type name

### LINKED ISSUE #120595: rustdoc: Long type names result in weird-looking documentation
Crates with long type names result in a strange-looking error, see [https://docs.rs/async-stripe/0.31.0/stripe/](https://docs.rs/async-stripe/0.31.0/stripe/)

image

I've tested this on Chrome and Firefox, and by building the docs locally from [https://github.com/arlyon/async-stripe](https://github.com/arlyon/async-stripe)

I believe the text is rendering vertically due to long type names like [https://docs.rs/async-stripe/0.31.0/stripe/enum.PaymentFlowsPrivatePaymentMethodsCardDetailsApiResourceEnterpriseFeaturesIncrementalAuthorizationIncrementalAuthorizationStatus.html](https://docs.rs/async-stripe/0.31.0/stripe/enum.PaymentFlowsPrivatePaymentMethodsCardDetailsApiResourceEnterpriseFeaturesIncrementalAuthorizationIncrementalAuthorizationStatus.html) which are auto-generated from OpenAPI.

Link to original issue [https://github.com/arlyon/async-stripe/issues/488](https://github.com/arlyon/async-stripe/issues/488)

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes https://github.com/rust-lang/rust/issues/120595.

Takeover of #126209.

cc @BradMarr
r? @notriddle

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Weird, different size locally vs in CI. Oh well...

**Review by notriddle — CHANGES_REQUESTED** [review]
Okay, it's really, really important to include a screenshot on a change that's this drastic:

Don't like it.

image

Also:

* If the text is going to be aligned to the right, it should be consistently aligned to the right. Notice, in the above screenshot, `array` is ragged-right, but `slice` is ragged-left.

* Even if it is consistent, that's merely better. It's not actually *good*. Large amounts of English, like these descriptions, should always be written left-to-right.

**Comment by the8472** [comment]
https://github.com/rust-lang/rust/issues/120595#issuecomment-2093833131 cites table layout intrinsic width issues.

I'm not sure if that's correct. The issue seems to be that the table columns have different `overflow-wrap` rules. If *both* are set to `anywhere` then it looks mostly fine(-ish). It could use some polish to indicate that wrapping happened, a dash or an indent on.

image

**Comment by GuillaumeGomez** [comment]
Finally took the time to update. I reverted back to the table display but with some small differences: there is now a padding at the bottom to make it easier to spot which item's short doc we're reading and also set the width of the items name fixed (to 50% of the parent) so that whatever their size, the display remains the same. Screenshots:

Screenshot from 2024-07-17 14-27-33
Screenshot from 2024-07-17 14-27-52
Screenshot from 2024-07-17 14-29-52

**Comment by GuillaumeGomez** [comment]
I'll add GUI tests once we agree with the UI changes.

@rustbot ready

**Comment by GuillaumeGomez** [comment]
If one item is super long and the others are "normal", then the rendering is the same. One thing I dislike with the current display is that the position of the short documentation changes depending on the width of the biggest item name. That's why I decided to go for fixed width (with still handling too long name).

**Comment by GuillaumeGomez** [comment]
That's a good point. Gonna add UI tests to this new size then.

**Comment by GuillaumeGomez** [comment]
Reduced the width of items name to 33% and added a GUI test.

File: src/librustdoc/html/static/css/rustdoc.css [hunk 1]
```diff
@@ -2174,6 +2174,14 @@ in src-script.js and main.js
 		padding: 2px 4px;
 		box-shadow: 0 0 4px var(--main-background-color);
 	}
+
+	.item-table > li > .item-name {
+		width: 33%;
+	}
+	.item-table > li > div {
+		padding-bottom: 5px;
```
Comment (cynecx) [inline src/librustdoc/html/static/css/rustdoc.css:2182]: Maybe it's just me but this makes the spacing in the y-axis look so "wasteful" and a bit unnatural. Is this really necessary? Or why can't we keep the old behavior.

**Pull request merged**